bootstrap/controller/commonapi.go

```diff
@@ -0,0 +1,217 @@
+	lc := container.LoggingClientFrom(dic.Get)
+	secretProvider := container.SecretProviderExtFrom(dic.Get)
+	authenticationHook := handlers.AutoConfigAuthenticationFunc(secretProvider, lc)
+	configuration := container.ConfigurationFrom(dic.Get)
+	c := CommonController{
+		dic:         dic,
+		serviceName: serviceName,
+		router:      r,
+		lc:          lc,
+		version: version{
+			serviceVersion: serviceVersion,
+		},
+		config: config{
+			configuration: configuration,
+		},
+	}
+	r.HandleFunc(common.ApiPingRoute, c.Ping).Methods(http.MethodGet) // Health check is always unauthenticated
+	r.HandleFunc(common.ApiVersionRoute, authenticationHook(c.Version)).Methods(http.MethodGet)
+	r.HandleFunc(common.ApiConfigRoute, authenticationHook(c.Config)).Methods(http.MethodGet)
+	r.HandleFunc(common.ApiSecretRoute, authenticationHook(c.AddSecret)).Methods(http.MethodPost)
+
+	return &c
+}
+
+// SetSDKVersion sets the service's skd version
+func (c *CommonController) SetSDKVersion(sdkVersion string) {
+	c.version.sdkVersion = sdkVersion
+}
+
+// SetCustomConfigInfo sets the custom configuration, which is used to include the service's custom config in the /config endpoint response.
+func (c *CommonController) SetCustomConfigInfo(customConfig interfaces.UpdatableConfig) {
+	c.config.customConfig = customConfig
+}
+
+// Ping handles the request to /ping endpoint. Is used to test if the service is working
+// It returns a response as specified by the API swagger in the openapi directory
+func (c *CommonController) Ping(writer http.ResponseWriter, request *http.Request) {
+	response := commonDTO.NewPingResponse(c.serviceName)
+	c.sendResponse(writer, request, response, http.StatusOK)
+}
+
+// Version handles the request to /version endpoint. Is used to request the service's versions
+// It returns a response as specified by the API swagger in the openapi directory
+func (c *CommonController) Version(writer http.ResponseWriter, request *http.Request) {
+	var response interface{}
+	if c.version.sdkVersion != "" {
+		response = commonDTO.NewVersionSdkResponse(c.version.serviceVersion, c.version.sdkVersion, c.serviceName)
+	} else {
+		response = commonDTO.NewVersionResponse(c.version.serviceVersion, c.serviceName)
+	}
+	c.sendResponse(writer, request, response, http.StatusOK)
+}
+
+// Config handles the request to /config endpoint. Is used to request the service's configuration
+// It returns a response as specified by the swagger in openapi/common
+func (c *CommonController) Config(writer http.ResponseWriter, request *http.Request) {
+	var fullConfig interface{}
+
+	if c.config.customConfig == nil {
+		// case of no custom configs
+		fullConfig = c.config.configuration
+	} else {
+		// create a struct combining the common configuration and custom configuration sections
+		fullConfig = struct {
+			interfaces.Configuration
+			CustomConfiguration interfaces.UpdatableConfig
+		}{
+			c.config.configuration,
+			c.config.customConfig,
+		}
```

Comment:
to make sure the response is the same as before, we need to convert the interface to map first
```
fullConfig := make(map[string]any)
mapstructure.Decode(c.config.configuration, &m)
fullConfig ["CustomConfiguration"]=c.config.customConfig
```